tests/integration/test_writes.py

```diff
@@ -778,3 +778,31 @@ def test_inspect_snapshots(
                 continue
 
             assert left == right, f"Difference in column {column}: {left} != {right}"
+
+
+@pytest.mark.integration
+def test_write_within_transaction(spark: SparkSession, session_catalog: Catalog, arrow_table_with_null: pa.Table) -> None:
```

Comment:
Shall we cover overwrite in the test as well?

Reply:
Sure thing :)